lib/ansible/plugins/connection/local.py

```diff
@@ -80,7 +80,7 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
 
         p = subprocess.Popen(
             cmd,
-            shell=isinstance(cmd, basestring),
+            shell=isinstance(cmd, (text_type, binary_type)),
```

Comment:
We could remove text_type here if we wanted.... binary_type is the important one to keep.
subprocess deals in byte strings internally.  Passing in text strings can yield tracebacks in both python2 and python3 in some circumstances (more often in pyhon2 than python3 but try this: so we try always to give bytes to subprocess.  Not sure if we want to remove text_type or not... it could work for many customers who are using just ascii or have good locales set if we leave it in.  OTOH, if we take it out, it could be an early warning for us that we have to adjust our  code somewhere.

Example of subprocess failure on python3 with text type:
```
LC_ALL=C python3  
>>> import subprocess
>>> subprocess.Popen('printf \u304f\u3089\u3068\u307f', shell=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.4/subprocess.py", line 859, in __init__
    restore_signals, start_new_session)
  File "/usr/lib64/python3.4/subprocess.py", line 1401, in _execute_child
    restore_signals, start_new_session, preexec_fn)
UnicodeEncodeError: 'ascii' codec can't encode characters in position 7-10: ordinal not in range(128)
```